internal/config/identity/openid/provider/keycloak.go

```diff
@@ -83,11 +84,17 @@ func (k *KeycloakProvider) LoginWithClientID(clientID, clientSecret string) erro
 
 // LookupUser lookup user by their userid.
 func (k *KeycloakProvider) LookupUser(userid string) (User, error) {
-	lookupUserID := k.adminURL + "/realms" + k.realm + "/users/" + userid
-	req, err := http.NewRequest(http.MethodGet, lookupUserID, nil)
+	userURL, err := url.Parse(k.adminURL)
 	if err != nil {
 		return User{}, err
 	}
+	userURL.Path = path.Join(userURL.Path, "realms", k.realm, "users", userid)
```

Comment:
@hellivan this is done automatically by http.NewRequest() not sure why we need to do this again and again NOTE: we are leaking memory here doing url.Parse() again and again. 

http.NewRequest() parses the input URL again.